stem/output.py

```diff
@@ -0,0 +1,255 @@
+    Abstract class for the definition of user output parameters (GiD, VTK, json).
+
+    Attributes:
+        -
+    """
+
+    pass
+
+    @abstractmethod
+    def assemble_parameters(self) -> Dict[str, Any]:
+        """
+        Abstract method for assembling the output properties into a nested dictionary
+        """
+        pass
+
+    @abstractmethod
+    def validate(self):
+        """
+        Abstract method for validating user inputs
+        """
+        pass
+
+    @abstractmethod
+    def is_output_process(self):
+        """
+        Abstract method for checking whether an output is in the output process list
+        (True for GiD, TVK) or not (False for JSON) which is in `processes`
+        """
+        pass
+
+
+@dataclass
+class GiDOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for GiD output
+
+    Attributes:
+        gidpost_flags (Dict[str, Any]):
+        file_label (str):
+        output_control_type (str):
+        output_interval (int):
+        body_output (bool):
+        node_output (bool):
+        skin_output (bool):
+        plane_output (List[str]):
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        point_data_configuration (List[str]):
+    """
+
+    gidpost_flags: Dict[str, Any] = field(default_factory=lambda: DEFAULT_GID_FLAGS)
+    file_label: str = "step"
+    output_control_type: str = "step"
+    output_interval: int = 1
+    body_output: bool = True
+    node_output: bool = False
+    skin_output: bool = False
+    plane_output: List[str] = field(default_factory=lambda: [])
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+    point_data_configuration: List[str] = field(default_factory=lambda: [])
+
+    def assemble_parameters(self) -> Dict[str, Any]:
+        """
+        Method for assembling the output properties for GiD format into a nested
+        dictionary.
+
+        Returns:
+            Dict[str, Any]: dictionary of a list containing the output parameters
+        """
+        return {
+            "postprocess_parameters": {
+                "result_file_configuration": dict(
+                    gidpost_flags=self.gidpost_flags,
+                    file_label=self.file_label,
+                    output_control_type=self.output_control_type,
+                    output_interval=self.output_interval,
+                    body_output=self.body_output,
+                    node_output=self.node_output,
+                    skin_output=self.skin_output,
+                    plane_output=self.plane_output,
+                    nodal_results=self.nodal_results,
+                    gauss_point_results=self.gauss_point_results,
+                ),
+                "point_data_configuration": self.point_data_configuration,
+            }
+        }
+
+    def validate(self):
+        pass
+
+    def is_output_process(self):
```

Comment:
i think the functionality of this function belongs in the Kratos_IO , as it is not a concern of the user how the projectparameters.json is built up.